src/urllib3/connection.py

```diff
@@ -369,6 +370,22 @@ def request_chunked(
         """
         self.request(method, url, body=body, headers=headers, chunked=True)
 
+    def getresponse(self):
+
+        # Get the response from http.client.HTTPConnection
+        httplib_response = super().getresponse()
+
+        # Wrap http.client.HTTPResponse as a urllib3.response.HTTPResponse
+        response = HTTPResponse.from_httplib(
+            httplib_response,
+            pool=self,
+            connection=response_conn,
+            retries=retries,
+            **response_kw,
+        ) # Post of these parameters will need to change since we are not calling from inside of urllib3 ConnectionPool anymore
```

Comment:
Most of these parameters from from ConnectionPool.urlopen:
https://github.com/urllib3/urllib3/blob/f9c7434090ce96ca93180137356531a9cfcce822/src/urllib3/connectionpool.py#L747-L753

When we move the wrapping to `HTTPConnection.getresponse()` we lose access to variables like `retries` and `pool`. 

It seems like `retries` might be useful as an instance variable? but I'm not sure about `pool`? Maybe it could be an optional argument like `get_response(pool=None)`.

I have not read the code for `from_httplib` but that is my next stop.